### PR TITLE
CORE-719: In BRGenericTransfer add a UIDS; Add genTransferEqual()

### PR DESCRIPTION
**Pull request by EBGToo**
When matching GEN BDB Transfers work hard to handle the ‘__fee__’.

Requires a Java update to `system.mergeTransfers()` and an argument addition to `cwmAnnounceGetTransferItemGEN()`

### CODE REVIEW

File: crypto/BRCryptoHash.c [hunk 1]
```diff
@@ -80,18 +80,27 @@ cryptoHashEqual (BRCryptoHash h1, BRCryptoHash h2) {
             : CRYPTO_FALSE);
 }
 
+static char *
+_cryptoHashAddPrefix (char *hash) {
+    char *result = malloc (strlen (hash) + 3);
+    strcpy (result, "0x");
+    strcat (result, hash);
+    free (hash);
```
Comment (michael-brd) [inline crypto/BRCryptoHash.c]: This (the free) seems like a scary side-effect to roll into the function.

Comment (EBGToo) [reply crypto/BRCryptoHash.c]: That is me obsessing on tail calls.  I pushed a change that at least makes it more explicit.

**Comment by EBGToo** [comment]
There is more work to do on this; specifically handing created GEN transfers.

**Comment by EBGToo** [comment]
I updated to handle a 'created transfers' that is then seen from a BDB recovery.  Notably, update the UIDS.

**Pull request merged**